setup.go

```diff
@@ -165,11 +173,41 @@ func setup(c *caddy.Controller) error {
 		}
 	}
 
-	fp.httpTransport.DialContext = (&net.Dialer{
-		Timeout:   fp.dialTimeout,
-		KeepAlive: 30 * time.Second,
-		DualStack: true,
-	}).DialContext
+	if fp.upstream != "" {
+		fixedURL, err := url.Parse(fp.upstream)
+		if err != nil {
+			return errors.New("wrong upstream address")
+		}
+		switch fixedURL.Scheme {
+		case "http":
+			fp.httpTransport.Proxy = http.ProxyURL(fixedURL)
```

Comment:
Why do we set this up only for http://upstream.com?
`fp.httpTransport` is used when forwardproxy gets http request for the actual destination, and does not seem to be related to upstream scheme.